tests/radio.spec.ts

```diff
@@ -0,0 +1,134 @@
+
+  await expect(opt4).toHaveClass(/bg-white/)
+  await expect(opt5).toHaveClass(/bg-white/)
+  await expect(opt6).toHaveClass(/bg-gray-9/)
+});
+
+
+test('Radio E2E Test, Label Position', async ({ page }) => {
+  await page.goto('/test.html');
+
+  // Label Default (Render Label On Top Of Radio) (Checks that button with text "Def 1" is below label of "Def Position") 
+  const def1Button = await page.locator("button:below(:text('Def Position'))").first().textContent()
+  expect(def1Button).toContain('Def 1')
+
+  // Label Top (Render Label On Top Of Radio) (Checks that button with text "Top 1" is below label of "Top Position")
+  const top1Button = await page.locator("button:below(:text('Top Position'))").first().textContent()
+  expect(top1Button).toContain('Top 1')
+  
+  // Label Left (Check That Button Is Right Of Label)
+  const left1Button = await page.locator("button:right-of(:text('Left Position'))").first().textContent()
+  expect(left1Button).toContain('Left 1')
+});
+
+
+test('Radio E2E Test, Tooltips', async ({ page }) => {
+  await page.goto('/test.html');
+
+  // Tooltip Default
+  const tooltipDef = page.getByTestId("radio-tooltip-def-test").locator('v-tooltip div').first()
+  await expect(tooltipDef).toHaveClass(/icon-info-outline/)
+
+  // Tooltip Warn
+  const tooltipWarn = page.getByTestId("radio-tooltip-warn-test").locator('v-tooltip div').first()
+  await expect(tooltipWarn).toHaveClass(/icon-error-outline text-warning-fg/)
+
+  // Tooltip Error
+  const tooltipError = page.getByTestId("radio-tooltip-error-test").locator('v-tooltip div').first()
+  await expect(tooltipError).toHaveClass(/icon-error-outline text-danger-fg/)
+
+  // Tooltip Info
+  const tooltipInfo = page.getByTestId("radio-tooltip-info-test").locator('v-tooltip div').first()
+  await expect(tooltipInfo).toHaveClass(/icon-info-outline/)
+
+  // Tooltip Left
+  const tooltipLeft = page.locator("div.icon-info-outline:left-of(:text('Tool 10'))").first()
+  await expect(tooltipLeft).toHaveClass(/icon-info-outline/)
+
+  // Tooltip Hover
+  const tooltipHover = page.getByTestId("radio-tooltip-hover-test").locator('v-tooltip div').first()
+  const tooltipText = page.getByRole('tooltip', { name: 'This is the hover tooltip test'})
+  await tooltipHover.hover()
+  await expect(tooltipText).toBeVisible()
+  await expect(tooltipText).toHaveText('This is the hover tooltip test')
+});
+
+
+test('Radio E2E Test, Readonly', async ({ page }) => {
+  await page.goto('/test.html');
+
+  // Readonly, Click
+  const readonly2 = page.getByRole('button', { name: 'Readonly 2' })
+  await expect(readonly2).toBeVisible()
+
+  const readonly2Selected = waitForCustomEventTimeout(page,'input')
+  await readonly2.press('Enter')
+  await readonly2Selected
+  
+  // Readonly Keydown
+  const readonly5 = page.getByRole('button', { name: 'Readonly 5' })
+  await expect(readonly5).toBeVisible()
+
+  const readonly5Selected = waitForCustomEventTimeout(page,'input')
```

Comment:
i think `waitForCustomEventTimeout` adds an event listener under the event `input` and returns true if it is received once. not sure what calling `waitForCustomEventTimeout` twice with the same event name would do (are two event listeners with the same name added? does the old one get deleted and reset?), would just double check to see that `readonly5selected` does indeed fail if the enter press on `readonly5` doesn't happen

Reply:
ah sorry ignore this i was thinking of `waitForCustomEvent`, didn't realize this was readonly